docs-compose/overwatch_docs/content/DataEngineer/Productionizing.md

```diff
@@ -55,6 +56,205 @@ created each run but it will be replaced on each subsequent run. If you would li
+val pipeline = "Bronze,Silver,Gold"
+val snapshotType = "Incremental"
+Snapshot.isValid(sourceETLDB,snapshotType,pipeline)
+```
+
+
+### Restore Process
+Restore is the reverse process of Snapshot.In data world the term Restore is very common. It simply means if our current working space is corrupted then we can restore the current workspace from the snapshot location we have used earlier.
+
+Same like Snapshot Restore can be run by both Databricks Job and Databricks Notebook.
+
+##### Restore through Databricks Notebook
+
+The code snippet to run the Restore process is as below:
+```scala
+import com.databricks.labs.overwatch.pipeline.Restore
+val sourcePrefix = "/mnt/overwatch_playground/721-snap/full"
+val targetPrefix = "/mnt/overwatch_playground/721-snap/restore"
+Restore.process(sourcePrefix,targetPrefix)
+```
+We need below parameters to run the Restore process through Databricks Notebook:
+
+| Param          | Type   | Optional  | Description                                                                              |
+|----------------|--------|-----------|------------------------------------------------------------------------------------------|
+| sourcePrefix   | String | No        | Source ETL Path Prefix from where restore need to be performed.                          |
+| targetPrefix   | String | No        | Target ETL Path Prefix to where restore data would be loaded.                            |
+
+After Restore process is done the tables from sourcePrefix will be restored in the `target prefix/globalshare` location
+
+![Restore](/images/DataEngineer/restore_path.png)
+
+##### Restore through Databricks Job
+Restore process can also be configured using Databricks Job. For this we need to run RestoreRunner Class.
+
+Below is the screenshot of the Job through which we run the snapshot process.
+
+1. **Type** - Jar
+2. **Main class** - com.databricks.labs.overwatch.pipeline.RestoreRunner
+3. **Parameters for the Job** - Same as mentioned above for `Restore through Databricks Notebook`
+
+![Restore_Job](/images/DataEngineer/restore_job.png)
+
+#### Validation Functionality
+This is a special module in Restore  Process. By default, it is part of Restore.process() but if user want to run it separately they can run it separately
+to check whether the configuration they would provide for Restore process, are correct or not.
+
+Below is the code snippet for the validation function
+
+```scala
+import com.databricks.labs.overwatch.pipeline.Restore
+val sourcePrefix = "/mnt/overwatch_playground/721-snap/full"
+val targetPrefix = "/mnt/overwatch_playground/721-snap/restore"
+Restore.isValid(sourcePrefix,targetPrefix)
+```
+
+
+### Migration Process
+Migration Process is same as Snapshot process with some added Functionality. Below are the steps involved in Migration Process:
+
+1. Stop Overwatch jobs  (Need to be done by User)
+2. Migrate all old data to new location. (Done by OW Migration Job) 
```

Comment:
can it will be like step wise like migration process.
Step-1) Stop the job
Step 2)Perform migration
run the below command to perform migration via notebook. 
As migration is a one time process i dont think job is needed to perform migration any thought???
Step-3)Resume OW job.

Reply:
Yes that's why I have mentioned in point 2,3 and 4 steps would be done by Overwatch Migration Process.

Also the job can be Optional. As for the technical PR we would need to create runner class for all Snapshot, Migration and Restore.